File: backend/shared/src/shared/schemas/dto/campaign_quality.py
from typing import List

from pydantic import BaseModel


class WorkerQualityDTO(BaseModel):
    workerId: str
    workerName: str
    workerAcronym: str
    numAvailableDays: int
    numDuties: float
    numOnCall: float
    timeWorkedMinutes: float
    timeWorkedWeekendMinutes: float
    numWorkingDays: int
    shiftDiversityScore: float
    shiftSpreadScore: float
    workTimeConsistencyScore: float
    individualScore: float
    fairnessScore: float


class CampaignQualityDTO(BaseModel):
    campaignId: str
    teamId: str
    startDate: float
    endDate: float
    numWorkers: int
    numWeeks: float
    fairnessDutiesScore: float
    fairnessOnCallScore: float
    fairnessTimeWorkedScore: float
    fairnessWeekendScore: float
    aggregateFairnessScore: float
    aggregateIndividualScore: float
    globalScore: float
    workers: List[WorkerQualityDTO]

File: frontend/src/types/campaignQuality.ts
export type WorkerQualityT = {
  workerId: string;
  workerName: string;
  workerAcronym: string;
  numAvailableDays: number;
  numDuties: number;
  numOnCall: number;
  timeWorkedMinutes: number;
  timeWorkedWeekendMinutes: number;
  numWorkingDays: number;
  shiftDiversityScore: number;
  shiftSpreadScore: number;
  workTimeConsistencyScore: number;
  individualScore: number;
  fairnessScore: number;
};

export type CampaignQualityT = {
  campaignId: string;
  teamId: string;
  startDate: number;
  endDate: number;
  numWorkers: number;
  numWeeks: number;
  fairnessDutiesScore: number;
  fairnessOnCallScore: number;
  fairnessTimeWorkedScore: number;
  fairnessWeekendScore: number;
  aggregateFairnessScore: number;
  aggregateIndividualScore: number;
  globalScore: number;
  workers: WorkerQualityT[];
};
